api/deps.py

```diff
@@ -0,0 +1,14 @@
+from typing import Generator
+
+from core.config import settings, SUPPORTED_DATA_PROVIDERS
+from remote.data_provider.data_provider import DataProvider
+
+
+def get_data_providers() -> Generator:
+    data_providers: set[DataProvider] = set()
+    stock_data_providers = settings.DATA_PROVIDERS
+    stock_data_providers = stock_data_providers.split(", ")
```

Comment:
using `stock_data_providers.split(", ")` because I'm expecting and environment variable with the following format

```
DATA_PROVIDERS = "YAHOO, GOOGLE"
```